### PR TITLE
added HasData() method to sqlstream/reader

### PR DESCRIPTION
**Pull request by mruediger**
this method allows to omit empty tables when pushing data to s3

### CODE REVIEW

File: sqlstream/reader.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ type DataRows interface {
 	Scan(dest ...interface{}) error
 }
 
-type sqlReader struct {
+type SQLReader struct {
```
Comment (omidaladini) [inline sqlstream/reader.go:19]: Any reason for upper casing this to make it public and visible from outside?

File: s5/s5.go [hunk 2]
```diff
@@ -57,6 +57,10 @@ func (s *S5) Run() error {
 		return err
 	}
 
+	if ! sqlReader.HasData() {
+		return nil
```
Comment (omidaladini) [inline s5/s5.go:61]: I'm thinking if the correct behaviour encountering an empty table is to create an empty file on S3. Empty table is in a valid state so the valid counter part would be an empty file.

File: sqlstream/reader.go (hunk 1 above)
Comment (mruediger) [inline sqlstream/reader.go:19]: You are right, it makes no sense since we are creating it via the NewSQLReader function.

File: sqlstream/reader.go [hunk 3]
```diff
@@ -114,6 +117,13 @@ func (r *sqlReader) deserializeData(rawColumns [][]byte) []string {
 	return tidyCols
 }
 
+func (r *SQLReader) HasData() bool {
+	if !r.cachedNext {
+		r.cachedNext = r.rows.Next()
+	}
+	return r.cachedNext
+}
+
```
Comment (omidaladini) [inline sqlstream/reader.go:126]: I feel a bit hesitant about adding `HasData` as [Reader.Read](https://golang.org/pkg/io/#Reader) already satisfies the purpose that HasData is supposed to serve. I understand it lets us know beforehand if there is any data, but we're just simulating "beforehand" by caching the next read (which adds a bit of unnecessary complexity).

Together with the previous comment (on correct behaviour of an empty table), perhaps we could fix the issue raised in this PR on `s5` package? Specifically by returning an error at s3Stream.UploadMultiPart saying that "you can't multipart upload empty data" and [recovering](https://github.com/omidaladini/s5/blob/master/s5/s5.go#L72) the error by PUTting an empty file. 

Basically, assigning responsibilities to respective packages.

File: s5/s5.go (hunk 2 above)
Comment (mruediger) [inline s5/s5.go:61]: If the specified query yielded no results, than creating a empty file on s3 would be the correct result. I was under the assumption though, that the underlying issue was, that we are trying to upload empty files. Therefore the whole point of the commit was to omit the upload if the result of a query yielded no results. As you stated in your mail, my assumption was wrong and I think we should drop this pull-request.

**Comment by omidaladini** [comment]
Thanks and let's fix this issue in another PR.
